cluster-autoscaler/scale_down.go

```diff
@@ -191,7 +190,7 @@ func ScaleDown(
 	}
 
 	// We look for only 1 node so new hints may be incomplete.
-	nodesToRemove, _, err := simulator.FindNodesToRemove(candidates, nodes, pods, client, predicateChecker, 1, false,
+	nodesToRemove, _, err := simulator.FindNodesToRemove(candidates, nodes, pods, client, predicateChecker, false,
```

Comment:
This code (FindNodesToRemove with fastDrain=false) does quite expensive manual verification that rs/rc/jobs/daemonset exist. It should not go over all nodes. So I would rather sort the candidates according to some criteria. And then get 1 (or more) that best fit the strategy.